pyproject.toml

```diff
@@ -6,32 +6,32 @@ requires = [
 ]
 build-backend = "setuptools.build_meta"
 
-[project]
+[tool.poetry]
 name = "gtfs-fetcher"
 version = "0.1.0"
 description = "Download, validate, and manage GTFS transit feeds"
-readme = "README.rst"
-requires-python = ">=3.10"
-license = {file = "LICENSE.txt"}
-authors = [
-  {email = "nils@gis-ops.com"}
-]
-maintainers = [
-  {name = "Nils Nolde", email = "nils@gis-ops.com"}
-]
+authors = ["Nils Nolde <nils@gis-ops.com>"]
+license = "MIT"
+readme = 'README.md'
 
-dependencies = [
-"prettytable >= 3.7.0, < 4.0.0",
-"requests >= 2.22.0, < 3.0.0",
-"typer[all] >= 0.9.0, < 1.0.0",
-"chalky >= 1.0.0, < 2.0.0",
-"halo >= 0.0.29, < 0.1.0"
-]
+[tool.poetry.dependencies]
+python = "^3.7.0"
+requests = "^2.22.0"
+typer = { version = "^0.9.0", extras = ["all"] }
+prettytable = "^3.7.0"
+halo = "^0.0.29"
 
-[project.optional-dependencies]
-tests = [
-  'pytest < 8',
-]
+[tool.poetry.dev-dependencies]
+pytest = "<8"
+mypy = "^1.3.0"
```

Comment:
great! but wouldn't we need to execute it somewhere?

Reply:
actually I think we can remove this too as a requirement and add it to `.pre-commit-config.yml` instead, which installs separately

Reply:
Yeah I would have to look into that....actually with mypy it's like we just have to add type hints and then it will generate warnings on its own seeing those hints....idk if there's a supposed to be a way to configure/execute it...no worries I'll check 👍🏼

Reply:
there's a lot of config around mypy in the pyproject.toml. I think it won't warn on "dynamic" typing (i.e. no type hints) by default.